packages/contracts-bedrock/src/L1/SystemConfig.sol

```diff
@@ -286,13 +296,35 @@ contract SystemConfig is OwnableUpgradeable, ISemver {
     /// @param _overhead New overhead value.
     /// @param _scalar   New scalar value.
     function _setGasConfig(uint256 _overhead, uint256 _scalar) internal {
+        require((uint256(0xff) << 248) & _scalar == 0, "SystemConfig: scalar exceeds max.");
+
         overhead = _overhead;
         scalar = _scalar;
 
         bytes memory data = abi.encode(_overhead, _scalar);
         emit ConfigUpdate(VERSION, UpdateType.GAS_CONFIG, data);
     }
 
+    /// @notice Updates gas config as of the Ecotone upgrade. Can only be called by the owner.
+    /// @param _basefeeScalar     New basefeeScalar value.
+    /// @param _blobbasefeeScalar New blobbasefeeScalar value.
+    function setGasConfigEcotone(uint32 _basefeeScalar, uint32 _blobbasefeeScalar) external onlyOwner {
+        _setGasConfigEcotone(_basefeeScalar, _blobbasefeeScalar);
+    }
+
+    /// @notice Internal function for updating the fee scalars as of the Ecotone upgrade.
+    /// @param _basefeeScalar     New basefeeScalar value.
+    /// @param _blobbasefeeScalar New blobbasefeeScalar value.
+    function _setGasConfigEcotone(uint32 _basefeeScalar, uint32 _blobbasefeeScalar) internal {
+        basefeeScalar = _basefeeScalar;
+        blobbasefeeScalar = _blobbasefeeScalar;
+
+        scalar = (uint256(0x01) << 248) | (uint256(_blobbasefeeScalar) << 32) | _basefeeScalar;
```

Comment:
Wondering if we explicitly also want to nullify the overhead, in case this is called on a pre-Ecotone-deployed contract. 
```suggestion
        overhead = 0;
        scalar = (uint256(0x01) << 248) | (uint256(_blobbasefeeScalar) << 32) | _basefeeScalar;
```

Reply:
I have a slight preference for not zeroing out the overhead as it gives us flexibility in passing arbitrary data into L2 if we ever need to do a "modification of interpretation of data" sort of upgrade like ecotone again, we can do this with a multicall of the legacy setter and the new setter so 2 config update events end up being emitted and we have complete control over the overhead